src/bridge-canister/src/runtime.rs

```diff
@@ -79,17 +82,60 @@ impl<Op: Operation> BridgeRuntime<Op> {
             self.scheduler.append_task(refresh_evm_params);
         }
 
-        let task_execution_result = self.scheduler.run(self.state.clone());
-
-        if let Err(err) = task_execution_result {
-            log::error!("task execution failed: {err}",);
+        if !self.state.borrow().should_process_operations() {
```

Comment:
I have big doubts about the way the scheduler is run after this change. With these changes applied:
1. Scheduler will run operations only after all the services are done with their operations. So we will not read logs EVM and schedule new operations during this time. Nevertheless, we add a new `collect_logs` and `refresh_evm_params` at every tick. so a bunch of them will be run one after another after the services are done with their operations.
2. We spawn a new async context to run this method (`run`), but then spawn a new async context to start operations, and then spawn a new context for each task. I don't see why the second spawn is necessary.
3. The tasks are run in parallel by the scheduler, e.g. `run_services` will return before any of the tasks was actually executed. It means that there's no meaning in separating services into the ones run before and after tasks. It would probably better to make sure the services can be run completely independent of the tasks (or even use tasks to run the services for better error handling and retry scheduling).

Reply:
> 1. Scheduler will run operations only after all the services are done with their operations. So we will not read logs EVM and schedule new operations during this time. Nevertheless, we add a new collect_logs and refresh_evm_params at every tick. so a bunch of them will be run one after another after the services are done with their operations.

Will be fixed in: https://infinityswap.atlassian.net/browse/EPROD-1005

> 2. We spawn a new async context to run this method (run), but then spawn a new async context to start operations, and then spawn a new context for each task. I don't see why the second spawn is necessary.

I don't think it is a big deal. We don't spawn context, only additional async task on inner IC async scheduler. It is extrimely lightweight and we have a lot of better variants if we want to provide some optimizations :)

> 3. The tasks are run in parallel by the scheduler...

I also thought about it... The "before" services work as expected: tasks start only when they are finished. The  "after" services run concurrently with tasks. We can solve it with some async signalling about task finish (e.g. async CondVar), but honestly I'm not sure it is necessary. I like to run "before" services first, because it is good to have:
- EVM logs collected right before the tasks are started. // Will be a service in next PR.
- EVM params refreshed right before the tasks are started. // Will be a service in next PR.

I think I can just rename the enum variant: `After` -> `Concurrent`, so service ordering will look like:
```rust
pub enum ServiceOrder {
    BeforeOperations,
    ConcurrentWithOperations,
}
```